### ISSUE TITLE
gummiboot is not compiled for OpenWrt 23.05

### ISSUE THREAD

**Issue by BKPepe** [openwrt/packages#22433]
Maintainer: @oskarirauta 
Environment: OpenWrt buildbots, 23.05, x86_x64

Description:

```
configure: WARNING: unrecognized options: --disable-nls

        gummiboot 48

        prefix:                  /usr
        arch:                    x86_64
        EFI machine type:        x64

        EFI libdir:              /builder/shared-workdir/build/sdk/staging_dir/target-x86_64_musl/usr/lib
        EFI ldsdir:              /builder/shared-workdir/build/sdk/staging_dir/target-x86_64_musl/usr/lib
        EFI includedir:          /builder/shared-workdir/build/sdk/staging_dir/target-x86_64_musl/usr/include

        blkid:                   yes
        man pages:               no

        QEMU:                    
        QEMU OVMF:               

make[4]: Entering directory '/builder/shared-workdir/build/sdk/build_dir/target-x86_64_musl/gummiboot-48.1'
  CC       src/setup/gummiboot-setup.o
  CC       src/setup/gummiboot-efivars.o
src/setup/efivars.c: In function 'efi_get_boot_option':
src/setup/efivars.c:354:39: warning: taking address of packed member of 'struct boot_option' may result in an unaligned pointer value [-Waddress-of-packed-member]
  354 |         title_size = utf16_size(header->title);
      |                                 ~~~~~~^~~~~~~
src/setup/efivars.c: In function 'efi_add_boot_option':
src/setup/efivars.c:471:24: warning: taking address of packed member of 'struct boot_option' may result in an unaligned pointer value [-Waddress-of-packed-member]
  471 |         to_utf16(option->title, title);
      |                  ~~~~~~^~~~~~~
src/setup/efivars.c:492:25: warning: taking address of packed member of 'struct device_path' may result in an unaligned pointer value [-Waddress-of-packed-member]
  492 |         to_utf16(devicep->path, path);
      |                  ~~~~~~~^~~~~~
src/setup/efivars.c:493:29: warning: taking address of packed member of 'struct device_path' may result in an unaligned pointer value [-Waddress-of-packed-member]
  493 |         tilt_slashes(devicep->path);
      |                      ~~~~~~~^~~~~~
  CCLD     gummiboot
make[4]: Leaving directory '/builder/shared-workdir/build/sdk/build_dir/target-x86_64_musl/gummiboot-48.1'
make[4]: Entering directory '/builder/shared-workdir/build/sdk/build_dir/target-x86_64_musl/gummiboot-48.1'
  CC       src/efi/util.o
In file included from src/efi/util.c:21:
src/efi/util.h: In function 'yes_no':
src/efi/util.h:27:27: warning: returning 'short unsigned int *' from a function with incompatible return type 'const CHAR16 *' {aka 'const int *'} [-Wincompatible-pointer-types]
   27 |         return b ? L"yes" : L"no";
      |                ~~~~~~~~~~~^~~~~~~
src/efi/util.c: In function 'efivar_set_int':
src/efi/util.c:114:25: warning: passing argument 3 of 'UnicodeSPrint' from incompatible pointer type [-Wincompatible-pointer-types]
  114 |         SPrint(str, 32, L"%d", i);
      |                         ^~~~~
      |                         |
      |                         short unsigned int *
In file included from src/efi/util.c:19:
/builder/shared-workdir/build/sdk/staging_dir/target-x86_64_musl/usr/include/efi/efilib.h:544:24: note: expected 'const CHAR16 *' {aka 'const int *'} but argument is of type 'short unsigned int *'
  544 |     IN CONST CHAR16   *fmt,
src/efi/util.c: In function 'efivar_set_time_usec':
src/efi/util.c:179:25: warning: passing argument 3 of 'UnicodeSPrint' from incompatible pointer type [-Wincompatible-pointer-types]
  179 |         SPrint(str, 32, L"%ld", usec);
      |                         ^~~~~~
      |                         |
      |                         short unsigned int *
/builder/shared-workdir/build/sdk/staging_dir/target-x86_64_musl/usr/include/efi/efilib.h:544:24: note: expected 'const CHAR16 *' {aka 'const int *'} but argument is of type 'short unsigned int *'
  544 |     IN CONST CHAR16   *fmt,
src/efi/util.c: In function 'utf8_to_16':
src/efi/util.c:188:21: warning: comparison is always true due to limited range of data type [-Wtype-limits]
  188 |         if (stra[0] < 0x80)
      |                     ^
  CC       src/efi/console.o
In file included from src/efi/console.c:21:
src/efi/util.h: In function 'yes_no':
src/efi/util.h:27:27: warning: returning 'short unsigned int *' from a function with incompatible return type 'const CHAR16 *' {aka 'const int *'} [-Wincompatible-pointer-types]
   27 |         return b ? L"yes" : L"no";
      |                ~~~~~~~~~~~^~~~~~~
src/efi/console.c: At top level:
src/efi/console.c:24: warning: "EFI_SIMPLE_TEXT_INPUT_EX_PROTOCOL_GUID" redefined
   24 | #define EFI_SIMPLE_TEXT_INPUT_EX_PROTOCOL_GUID \
      | 
In file included from /builder/shared-workdir/build/sdk/staging_dir/target-x86_64_musl/usr/include/efi/efi.h:67,
                 from src/efi/console.c:18:
/builder/shared-workdir/build/sdk/staging_dir/target-x86_64_musl/usr/include/efi/eficonex.h:22: note: this is the location of the previous definition
   22 | #define EFI_SIMPLE_TEXT_INPUT_EX_PROTOCOL_GUID \
      | 
src/efi/console.c:29:29: error: conflicting types for 'EFI_INPUT_RESET_EX'; have 'EFI_STATUS (*)()' {aka 'long unsigned int (*)()'}
   29 | typedef EFI_STATUS (EFIAPI *EFI_INPUT_RESET_EX)(
      |                             ^~~~~~~~~~~~~~~~~~
/builder/shared-workdir/build/sdk/staging_dir/target-x86_64_musl/usr/include/efi/eficonex.h:61:10: note: previous declaration of 'EFI_INPUT_RESET_EX' with type 'EFI_INPUT_RESET_EX' {aka 'long unsigned int (*)(struct _EFI_SIMPLE_TEXT_INPUT_EX_PROTOCOL *, unsigned char)'}
   61 | (EFIAPI *EFI_INPUT_RESET_EX) (
      |          ^~~~~~~~~~~~~~~~~~
src/efi/console.c:39:3: error: conflicting types for 'EFI_KEY_STATE'; have 'struct <anonymous>'
   39 | } EFI_KEY_STATE;
      |   ^~~~~~~~~~~~~
/builder/shared-workdir/build/sdk/staging_dir/target-x86_64_musl/usr/include/efi/eficonex.h:32:3: note: previous declaration of 'EFI_KEY_STATE' with type 'EFI_KEY_STATE'
   32 | } EFI_KEY_STATE;
      |   ^~~~~~~~~~~~~
src/efi/console.c:44:3: error: conflicting types for 'EFI_KEY_DATA'; have 'struct <anonymous>'
   44 | } EFI_KEY_DATA;
      |   ^~~~~~~~~~~~
/builder/shared-workdir/build/sdk/staging_dir/target-x86_64_musl/usr/include/efi/eficonex.h:37:3: note: previous declaration of 'EFI_KEY_DATA' with type 'EFI_KEY_DATA'
   37 | } EFI_KEY_DATA;
      |   ^~~~~~~~~~~~
src/efi/console.c:46:29: error: redefinition of typedef 'EFI_INPUT_READ_KEY_EX' with different type
   46 | typedef EFI_STATUS (EFIAPI *EFI_INPUT_READ_KEY_EX)(
      |                             ^~~~~~~~~~~~~~~~~~~~~
/builder/shared-workdir/build/sdk/staging_dir/target-x86_64_musl/usr/include/efi/eficonex.h:68:10: note: previous declaration of 'EFI_INPUT_READ_KEY_EX' with type 'EFI_INPUT_READ_KEY_EX' {aka 'long unsigned int (*)(struct _EFI_SIMPLE_TEXT_INPUT_EX_PROTOCOL *, EFI_KEY_DATA *)'}
   68 | (EFIAPI *EFI_INPUT_READ_KEY_EX) (
      |          ^~~~~~~~~~~~~~~~~~~~~
src/efi/console.c:51:29: error: redefinition of typedef 'EFI_SET_STATE' with different type
   51 | typedef EFI_STATUS (EFIAPI *EFI_SET_STATE)(
      |                             ^~~~~~~~~~~~~
/builder/shared-workdir/build/sdk/staging_dir/target-x86_64_musl/usr/include/efi/eficonex.h:75:10: note: previous declaration of 'EFI_SET_STATE' with type 'EFI_SET_STATE' {aka 'long unsigned int (*)(struct _EFI_SIMPLE_TEXT_INPUT_EX_PROTOCOL *, unsigned char *)'}
   75 | (EFIAPI *EFI_SET_STATE) (
      |          ^~~~~~~~~~~~~
src/efi/console.c:56:29: error: redefinition of typedef 'EFI_KEY_NOTIFY_FUNCTION' with different type
   56 | typedef EFI_STATUS (EFIAPI *EFI_KEY_NOTIFY_FUNCTION)(
      |                             ^~~~~~~~~~~~~~~~~~~~~~~
/builder/shared-workdir/build/sdk/staging_dir/target-x86_64_musl/usr/include/efi/eficonex.h:82:10: note: previous declaration of 'EFI_KEY_NOTIFY_FUNCTION' with type 'EFI_KEY_NOTIFY_FUNCTION' {aka 'long unsigned int (*)(EFI_KEY_DATA *)'}
   82 | (EFIAPI *EFI_KEY_NOTIFY_FUNCTION) (
      |          ^~~~~~~~~~~~~~~~~~~~~~~
src/efi/console.c:60:29: error: redefinition of typedef 'EFI_REGISTER_KEYSTROKE_NOTIFY' with different type
   60 | typedef EFI_STATUS (EFIAPI *EFI_REGISTER_KEYSTROKE_NOTIFY)(
      |                             ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/builder/shared-workdir/build/sdk/staging_dir/target-x86_64_musl/usr/include/efi/eficonex.h:88:10: note: previous declaration of 'EFI_REGISTER_KEYSTROKE_NOTIFY' with type 'EFI_REGISTER_KEYSTROKE_NOTIFY' {aka 'long unsigned int (*)(struct _EFI_SIMPLE_TEXT_INPUT_EX_PROTOCOL *, EFI_KEY_DATA *, long unsigned int (*)(EFI_KEY_DATA *), void **)'}
   88 | (EFIAPI *EFI_REGISTER_KEYSTROKE_NOTIFY) (
      |          ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~
src/efi/console.c:67:29: error: redefinition of typedef 'EFI_UNREGISTER_KEYSTROKE_NOTIFY' with different type
   67 | typedef EFI_STATUS (EFIAPI *EFI_UNREGISTER_KEYSTROKE_NOTIFY)(
      |                             ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/builder/shared-workdir/build/sdk/staging_dir/target-x86_64_musl/usr/include/efi/eficonex.h:97:10: note: previous declaration of 'EFI_UNREGISTER_KEYSTROKE_NOTIFY' with type 'EFI_UNREGISTER_KEYSTROKE_NOTIFY' {aka 'long unsigned int (*)(struct _EFI_SIMPLE_TEXT_INPUT_EX_PROTOCOL *, void *)'}
   97 | (EFIAPI *EFI_UNREGISTER_KEYSTROKE_NOTIFY) (
      |          ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
src/efi/console.c:72:16: error: redefinition of 'struct _EFI_SIMPLE_TEXT_INPUT_EX_PROTOCOL'
   72 | typedef struct _EFI_SIMPLE_TEXT_INPUT_EX_PROTOCOL {
      |                ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/builder/shared-workdir/build/sdk/staging_dir/target-x86_64_musl/usr/include/efi/eficonex.h:102:16: note: originally defined here
  102 | typedef struct _EFI_SIMPLE_TEXT_INPUT_EX_PROTOCOL{
      |                ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
src/efi/console.c:79:3: error: conflicting types for 'EFI_SIMPLE_TEXT_INPUT_EX_PROTOCOL'; have 'struct _EFI_SIMPLE_TEXT_INPUT_EX_PROTOCOL'
   79 | } EFI_SIMPLE_TEXT_INPUT_EX_PROTOCOL;
      |   ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/builder/shared-workdir/build/sdk/staging_dir/target-x86_64_musl/usr/include/efi/eficonex.h:109:3: note: previous declaration of 'EFI_SIMPLE_TEXT_INPUT_EX_PROTOCOL' with type 'EFI_SIMPLE_TEXT_INPUT_EX_PROTOCOL' {aka 'struct _EFI_SIMPLE_TEXT_INPUT_EX_PROTOCOL'}
  109 | } EFI_SIMPLE_TEXT_INPUT_EX_PROTOCOL;
      |   ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
make[4]: *** [Makefile:1096: src/efi/console.o] Error 1
make[4]: Leaving directory '/builder/shared-workdir/build/sdk/build_dir/target-x86_64_musl/gummiboot-48.1'
```

Snippet taken from https://downloads.openwrt.org/releases/faillogs-23.05/x86_64/packages/gummiboot/compile.txt

**Comment by jefferyto** [comment]
gummiboot is failing to build for master as well.

Perhaps it would be worth looking into whether systemd-boot can be built and packaged separate from the rest of systemd.

**Comment by 1715173329** [comment]
Should be fixed by #22696

**Comment by jefferyto** [comment]
Fixed by #22696 (339360045328a3338c08dfe39e6105dac886bf4f, 625efd9671721a5df6d867b737899cf6dc1beb26).
